src/Raven.Server/ServerWide/Commands/Subscriptions/AcknowledgeSubscriptionBatchCommand.cs

```diff
@@ -60,10 +66,60 @@ protected override BlittableJsonReaderObject GetUpdatedValue(long index, RawData
+                {
+                    if (string.IsNullOrEmpty(subscription.ChangeVectorForNextBatchStartingPoint))
+                    {
+                        // can update
+                    }
+                    else
+                    {
+                        var currentLocalShardCv = ShardLocalChangeVector.ToChangeVector();
+                        var currentCvInStorage = subscription.ChangeVectorForNextBatchStartingPoint.ToChangeVector();
+
+                        foreach (var entry in currentLocalShardCv)
+                        {
+                            if (currentCvInStorage.Any(x => x.DbId == entry.DbId))
+                                throw new SubscriptionChangeVectorUpdateConcurrencyException($"Can't acknowledge sharded subscription with name '{subscriptionName}' due to inconsistency in change vector progress. Probably there was an admin intervention that changed the change vector value. Stored value: {subscription.ChangeVectorForNextBatchStartingPoint}, received value: {LastKnownSubscriptionChangeVector}");
+                        }
+                    }
+                }
+                else
+                {
+                    if (string.IsNullOrEmpty(subscription.ChangeVectorForNextBatchStartingPoint))
+                    {
+                        throw new SubscriptionChangeVectorUpdateConcurrencyException($"Can't acknowledge sharded subscription with name '{subscriptionName}' due to inconsistency in change vector progress. Probably there was an admin intervention that changed the change vector value. Stored value: {subscription.ChangeVectorForNextBatchStartingPoint}, received value: {LastKnownSubscriptionChangeVector}");
+                    }
+                    else
+                    {
+                        var currentCvInStorage = subscription.ChangeVectorForNextBatchStartingPoint.ToChangeVector();
+                        var currentCvInCommand = LastKnownSubscriptionChangeVector.ToChangeVector();
+                        var currentLocalShardCv = ShardLocalChangeVector.ToChangeVector();
+
+                        foreach (var entry in currentLocalShardCv)
+                        {
+                            var lastStorageCv = currentCvInStorage.FirstOrDefault(x => x.DbId == entry.DbId);
+                            var lastCommandCv = currentCvInCommand.FirstOrDefault(x => x.DbId == entry.DbId);
+                            if (lastStorageCv.Equals(lastCommandCv) == false)
+                            {
+                                throw new SubscriptionChangeVectorUpdateConcurrencyException($"Can't acknowledge sharded subscription with name '{subscriptionName}' due to inconsistency in change vector progress. Probably there was an admin intervention that changed the change vector value. Stored value: {subscription.ChangeVectorForNextBatchStartingPoint}, received value: {LastKnownSubscriptionChangeVector}");
+                            }
```

Comment:
what are you doing here?

Reply:
Need an answer

Reply:
for each database id of shard's local change vector, I check if the sub change vector for this database id (if the shard has replication factor > 1 it might have few db ids) is the same for both `LastKnownSubscriptionChangeVector` and `subscription.ChangeVectorForNextBatchStartingPoint` (the last know cv for ack and current cv in the storage, which are merged cv of all shards now)